src/circuit/gadget/poseidon/pow5t3.rs

```diff
@@ -143,18 +147,24 @@ impl<F: FieldExt> Pow5T3Chip<F> {
             };
 
             vec![
-                s_partial.clone() * (pow_5(cur_0 + rc_a0) - mid_0.clone()),
-                s_partial.clone()
-                    * (pow_5(
-                        mid_0.clone() * m_reg[0][0]
-                            + (cur_1.clone() + rc_a1.clone()) * m_reg[0][1]
-                            + (cur_2.clone() + rc_a2.clone()) * m_reg[0][2]
-                            + rc_b0,
-                    ) - (next_0.clone() * m_inv[0][0]
-                        + next_1.clone() * m_inv[0][1]
-                        + next_2.clone() * m_inv[0][2])),
-                partial_round_linear(1, rc_b1),
-                partial_round_linear(2, rc_b2),
```

Comment:
Refactor this to use a map that multiplies by `s_partial`.

Reply:
Will do in a separate PR.